Source/Plugins/EditorModules/SceneView/Modules/SceneView.cs

```diff
@@ -1477,16 +1477,31 @@ private void lockedToolStripMenuItem_Click(object sender, EventArgs e)
 
 		private void gameObjectToolStripMenuItem_Click(object sender, EventArgs e)
 		{
-			// Create the GameObject
-			GameObject obj = this.CreateGameObject(this.objectView.SelectedNode);
-			GameObjectNode objNode = this.FindNode(obj);
+			// Create new objects
+			List<GameObjectNode> newObjects = new List<GameObjectNode>();
+			if (this.objectView.SelectedNodes.Count <= 1)
+			{
+				GameObject obj = this.CreateGameObject(this.objectView.SelectedNode);
+				GameObjectNode objNode = this.FindNode(obj);
+				newObjects.Add(objNode);
+			}
+			else
+			{
+				foreach (TreeNodeAdv node in this.objectView.SelectedNodes)
+				{
+					GameObject obj = this.CreateGameObject(node);
+					GameObjectNode objNode = this.FindNode(obj);
+					newObjects.Add(objNode);
+				}
+			}
 
-			// Deselect previous
 			this.objectView.ClearSelection();
 
-			// Select new node
-			if (objNode != null)
+			// Select newly created objects
+			foreach (GameObjectNode objNode in newObjects)
```

Comment:
The new loop ensures all view items matching object nodes will be selected, which is exactly as expected. However, it also calls EnsureVisible and BeginEdit on all of them, even though only the last one will have a lasting effect. Can you adjust the code, so only the last (or first) view node gets those two calls?